code/modules/mob/living/basic/lavaland/lobstrosity/lobstrosity.dm

```diff
@@ -67,22 +67,33 @@
 /mob/living/basic/mining/lobstrosity/tamed(mob/living/tamer, obj/item/food)
 	new /obj/effect/temp_visual/heart(loc)
 	/// Pet commands for this mob, however you'll have to tame juvenile lobstrosities to a trained adult one.
-	var/static/list/pet_commands = list(
+	var/list/pet_commands = list(
 		/datum/pet_command/idle,
 		/datum/pet_command/free,
 		/datum/pet_command/point_targeting/attack,
+		charge_command,
 		/datum/pet_command/follow,
 		/datum/pet_command/point_targeting/fish,
 	)
-
+	if(!HAS_TRAIT(src, TRAIT_MOB_RELAY_HAPPINESS))
+		AddComponent(/datum/component/happiness)
```

Comment:
Why are we guarding this with a trait? Components can just be set to disregard duplicates, you would only need this for an element.